include/otbObiaLSBaatzSegmentationScheduler.txx

```diff
@@ -790,140 +865,168 @@ LSBaatzSegmentationScheduler<TInputImage>
+        //}
 
-        }
+	// sending the size of the graph
+
+	std::cout << " PROCESSOR : " << mpiConfig->GetMyRank() << " , size of the graph : " << m_SerializedStabilityMargin.size() << std::endl;
+
+	unsigned long curSize = m_SerializedStabilityMargin.size();
 
-//        for(uint32_t ty = 0; ty < this->m_NumberOfTilesY; ty++)
-//        {
-//            for(uint32_t tx = 0; tx < this->m_NumberOfTilesX; tx++)
-//            {
-//                if(mpiTools->IsMyTurn(tid))
-//                {
-//                	// Serialize the current graph
-//                   m_SerializedStabilityMargin = GraphOperationsType::SerializeGraph(this->m_Graph);
-//
-//                }
-//
-//                tid++;
-//
-//            } // end for(uint32_t tx = 0; tx < nbTilesX; tx++)
-//
-//        } // end for(uint32_t ty = 0; ty < nbTilesY; ty++)
+	MPI_Send(&curSize,
+		 1,
+		 MPI_UNSIGNED_LONG,
+		 0,
+		 0,
+		 MPI_COMM_WORLD);
 
     } // end if (rank !=0 )
 
     /* Synchronization point */
     mpiConfig->barrier();
 
-    m_MaxNumberOfBytes = m_SerializedStabilityMargin.size();
+    m_MaxNumberOfBytes = m_SerializedStabilityMargin.size(); // BDU : useless now
 
     mpiTools->ComputeMax<unsigned long int>(m_MaxNumberOfBytes, MPI_UNSIGNED_LONG);
 
+    std::cout << "PROC : " << mpiConfig->GetMyRank() << " m_MaxNumberOfBytes " << m_MaxNumberOfBytes <<std::endl;  // BDU : anyway ...
 
     if(mpiConfig->GetMyRank() == 0)
     {
         // Simultaneous graph transfer to the master process.
         MPI_Request requests[mpiConfig->GetNbProcs() - 1];
         MPI_Status statuses[mpiConfig->GetNbProcs() - 1];
-        std::vector< std::vector<char> > serializedOtherGraphs(mpiConfig->GetNbProcs() - 1, std::vector<char>(m_MaxNumberOfBytes, char()));
-        std::vector< int > numberOfRecvBytesPerGraph(mpiConfig->GetNbProcs()-1, 0);
+        //std::vector< std::vector<char> > serializedOtherGraphs(mpiConfig->GetNbProcs() - 1, std::vector<char>(m_MaxNumberOfBytes, char()));
+	//std::vector< int > numberOfRecvBytesPerGraph(mpiConfig->GetNbProcs()-1, 0);        
 
+	std::vector< std::vector<char> > serializedOtherGraphs;
+	serializedOtherGraphs.reserve(mpiConfig->GetNbProcs() - 1);
+	
         for(unsigned int r = 1; r < mpiConfig->GetNbProcs(); r++)
         {
-            serializedOtherGraphs[r-1].assign(m_MaxNumberOfBytes, char());
-            MPI_Irecv(&(serializedOtherGraphs[r-1][0]),
-                      m_MaxNumberOfBytes,
-                      MPI_CHAR,
-                      r,
-                      MPI_ANY_TAG, 
-                      MPI_COMM_WORLD, 
-                      &(requests[r-1]));
+	   
+            //serializedOtherGraphs[r-1].assign(m_MaxNumberOfBytes, char());
+	    serializedOtherGraphs.push_back(std::vector<char>(sizePerGraph[r-1]));
+
+	    // BDU
+	    std::cout << "MASTER, receiving data from proc " << r <<std::endl;
+	
+	    if(sizePerGraph[r-1] > 2147483647)
```

Comment:
2147483647 : avoid magic numbers (derive the value from constant or add a comment to explain where it comes from)

Reply:
I did not understood this magic number thing?

Reply:
Do you mean what are magic numbers or why there is one here ?

Reply:
I don't get how it is used here. It's the maximum value we can do with a 32-bits int...

Reply:
MPI_Send routine can not accept an array of data longer than max 32-bits int (since this is the type of the array lenght parameter). On the other, our internal data structure based on std::vector can. In some extreme case with huge graphs, it might happen that the graph to send is longer than this maximum lenght MPI send/recieve. To avoid crashes in this case, we check this condition and split the comm if needed.